File: python/数组/414. 第三大的数.py
from typing import List
class Solution:
    def thirdMax(self, nums: List[int]) -> int:
        MIN = float('-inf')
        a,b,c = MIN,MIN,MIN
        for i in nums:
            if i == a or i == b or i == c:
                continue
            if i > a:
                a,b,c=i,a,b
            elif i > b:
                b,c = i,b
            elif i > c:
                c = i
        if c == MIN:
            return a
        return c

File: go/数组/414. 第三大的数.go
package array

func thirdMax(nums []int) int {
	var MIN = -1 << 63
	a, b, c := MIN, MIN, MIN

	for _, n := range nums {
		if n == a || n == b || n == c {
			continue
		}

		if n > a {
			a, b, c = n, a, b
		} else if n > b {
			b, c = n, b
		} else if n > c {
			c = n
		}
	}

	if c == MIN {
		return a
	}
	return c
}
